app/src/main/java/com/example/scoutconcordia/ShuttleInfo.java

```diff
@@ -0,0 +1,542 @@
+                }
+
+                messageToUser = "The current time is: " + timeString + " (" + timeOfDay + ")" + " and the next bus shuttle comes at: " + schedule_MonToThurs_SGW[indexPosition] + " (" + nextShuttleTime + ")";
+
+                break;
+
+            case "Friday":
+
+                shuttleTimes = getSGWFridayTimes();
+
+                for (int i = 0; i < shuttleTimes.length; i++){
+                    if (timeOfDay < shuttleTimes[i]){
+                        nextShuttleTime = shuttleTimes[i];
+                        indexPosition = i;
+                        break;
+                    }
+                }
+
+                messageToUser = "Today is " + today + ". The current time is: " + timeString + " (" + timeOfDay + ")" + " and the next bus shuttle comes at: " + schedule_friday_SGW[indexPosition] + " (" + nextShuttleTime + ")";
+
+                break;
+
+
+            case "Saturday":
+                messageToUser = "There are no bus shuttles on Saturdays!";
+                break;
+        }
+
+        // The shuttle time is hardcoded at 30 minutes, because of fluctuations, it is an estimation
+        double estimatedTime =  Math.round((((nextShuttleTime - timeOfDay)*60 ) + 30) * 100.0) / 100.0;
+
+        String overallEstimation = "The estimated route time is: " + estimatedTime;
+
+        System.out.println((messageToUser + "\n" + overallEstimation + " minutes"));
+
+//            return (messageToUser + "\n" + overallEstimation);
+
+        return estimatedTime;
+
+    }
+
+
+
+
+    // This method checks the time and day and finds the next relevant shuttle bus time, from Loyola.
+    // From there, it builds an estimate of how long it would take for the entire trip.
+
+    public double getNextEarliestTimeFromLoyola() {
+
+
+        Calendar cal = Calendar.getInstance();
+//            cal.setTime(date);
+        double hour = cal.get(Calendar.HOUR_OF_DAY);
+        double minutes = cal.get(Calendar.MINUTE);
+
+        String timeString = (int) hour + ":" + (int) minutes;
+
+        // This variable keeps track of the current time.
+        double timeOfDay =  Math.round((hour + (minutes / 60)) * 100.0) / 100.0;
+
+        int currentDay = Calendar.getInstance().get(Calendar.DAY_OF_WEEK);
+
+        // This variable keeps track of the current day.
+        String today = null;
+
+        switch(currentDay){
+
+            case 1:
+                today = "Sunday";
+                break;
+
+            case 2:
+                today = "Monday";
+                break;
+
+            case 3:
+                today =  "Tuesday";
+                break;
+
+            case 4:
+                today =  "Wednesday";
+                break;
+
+            case 5:
+                today =  "Thursday";
+                break;
+
+            case 6:
+                today =   "Friday";
+                break;
+
+            case 7:
+                today =  "Saturday";
+                break;
+        }
```

Comment:
In general, switch statements should always have a default case.

The default case could handle potential errors or unexpected inputs. What if currentDay was null? You should include a default case to handle these situations.

Reply:
Same thing for all the switch statements in this class.

Reply:
> Same thing for all the switch statements in this class.

Good catch, I don't know how I missed that. It is fixed in the latest push.